tests/mocha/workspace_test.js

```diff
@@ -704,4 +667,367 @@ suite('Workspace', function() {
       chai.assert.isNull(this.workspace.getBlockById(this.blockB));
     });
   });
+
+  suite('Undo/Redo', function() {
+    function temporary_fireEvent(event) {
+      if (!Blockly.Events.isEnabled()) {
+        return;
+      }
+      Blockly.Events.FIRE_QUEUE_.push(event);
+      Blockly.Events.fireNow_();
+    }
+
+    setup(function() {
+      this.savedFireFunc_ = Blockly.Events.fire;
+      Blockly.Events.fire = temporary_fireEvent;
+      temporary_fireEvent.firedEvents_ = [];
+    });
+
+    teardown(function() {
+      Blockly.Events.fire = this.savedFireFunc_;
+    });
+
+    function createTwoVarsDifferentTypes(workspace) {
+      workspace.createVariable('name1', 'type1', 'id1');
+      workspace.createVariable('name2', 'type2', 'id2');
+    }
+
+    suite('createVariable', function() {
+      test('Undo only', function() {
+        createTwoVarsDifferentTypes(this.workspace);
+
+        this.workspace.undo();
+        assertVariableValues(this.workspace, 'name1', 'type1', 'id1');
+        chai.assert.isNull(this.workspace.getVariableById('id2'));
+
+        this.workspace.undo();
+        chai.assert.isNull(this.workspace.getVariableById('id1'));
+        chai.assert.isNull(this.workspace.getVariableById('id2'));
+      });
+
+      test('Undo and redo', function() {
+        createTwoVarsDifferentTypes(this.workspace);
+
+        this.workspace.undo();
```

Comment:
Make sure that the variable was actually removed after the undo.

Reply:
The previous test tests that the variable was removed after the undo.

Reply:
Yeah, I still think it would be nice to have. If the other test gets removed or messed with in some way and doesn't catch this then yours will report that redo is working properly when it is not actually testing it. However, you do other checks later on in this test that would probably catch this, so up to you.

Reply:
Ok, I can see the value, I've updated the undo/redo tests.